classes/DepositPointDAO.inc.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/**
+ * @file plugins/generic/sword/classes/DepositPointDAO.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2003-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class DepositPointDAO
+ * @ingroup plugins_generic_sword_classes
+ *
+ * @brief Operations for retrieving and modifying DepositPoint objects.
+ */
+
+import('lib.pkp.classes.db.DAO');
+
+class DepositPointDAO extends DAO {
+	/** @var SwordPlugin reference to SWORD plugin */
+	protected $_plugin = null;
+
+	/**
+	 * Constructor
+	 * @param $parentPlugin SwordPlugin
+	 */
+	public function __construct(SwordPlugin $parentPlugin) {
```

Comment:
We're not currently using type declarations elsewhere, but I see no harm in starting to use them. I wonder whether Doxygen will pick them up from here so we don't have to declare them in two places...

Reply:
I'm not familiar with Doxygen's inner working. But let me know if I need to remove the type specification.